Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -1814,17 +1811,27 @@ LayoutRect RenderLayerScrollableArea::scrollRectToVisible(const LayoutRect& abso
         // For `direction: rtl; writing-mode: horizontal-{tb,bt}` and `writing-mode: vertical-rl`
         // boxes, the scroll bar is on the left side. The visible rect starts from the right side
         // of the scroll bar. So the x of localExposeRect should start from the same position too.
-        localExposeRect.moveBy(LayoutPoint(-verticalScrollbarWidth(), 0));
+        localExposeRect.moveBy(LayoutPoint(-verticalScrollbarWidth, 0));
     }
 
     // scroll-padding applies to the scroll container, but expand the rectangle that we want to expose in order
     // simulate padding the scroll container. This rectangle is passed up the tree of scrolling elements to
     // ensure that the padding on this scroll container is maintained.
-    LayoutRect layerBounds(0_lu, 0_lu, box->clientWidth(), box->clientHeight());
     localExposeRect.expand(box->scrollPaddingForViewportRect(layerBounds));
+    return localExposeRect;
+}
+
+LayoutRect RenderLayerScrollableArea::scrollRectToVisible(const LayoutRect& absoluteRect, const ScrollRectToVisibleOptions& options)
+{
+    RenderBox* box = layer().renderBox();
+    ASSERT(box);
+    
+    LayoutRect layerBounds(0_lu, 0_lu, box->clientWidth(), box->clientHeight());
 
-    auto revealRect = getRectToExposeForScrollIntoView(layerBounds, localExposeRect, options.alignX, options.alignY);
+    LayoutRect localExposeRect = getLocalExposeRect(absoluteRect, box, verticalScrollbarWidth(), layerBounds);
+    LayoutRect localVisiblityRect = getLocalExposeRect(options.visibilityRect, box, verticalScrollbarWidth(), layerBounds);
```

Comment:
Why do we compute `absoluteRect` and `options.visibilityRect` in absolute coordinates, only to convert them back to local coordinates here? That's error prone and in some cases involving 3d transforms might even not produce the same results.

Reply:
This is how the code was operating previously. I'm not sure if there's a good way to get the local rect directly without converting the absolute rect.